seismicpro/refractor_velocity/utils.py

```diff
@@ -91,3 +92,121 @@ def dump_refractor_velocities(refractor_velocities, path, encoding="UTF-8"):
+
+
+# pylint: disable-next=too-many-arguments
+def calc_optimal_velocity(offsets, times, init=None, bounds=None, min_velocity_step=400, min_refractor_size=400,
+                          loss="L1", huber_coef=20, min_refractors=1, max_refractors=10, find_weathering=False,
+                          debug=False):
+    """Calculate a near-surface velocity model with a number of refractors that give minimal loss.
+
+    Parameters
+    ----------
+    offsets : 1d ndarray
+        Offsets of traces. Measured in meters.
+    times : 1d ndarray
+        Time of first break for each trace. Measured in milliseconds.
+    init : dict, optional
+        Initial values of model parameters.
+    bounds : dict, optional
+        Lower and upper bounds of model parameters.
+    min_velocity_step : int, optional, defaults to 400
+        Minimum difference between velocities of two adjacent refractors.
+    min_refractor_size : int, optional, defaults to 400
+        Minimum offset range covered by each refractor.
+    loss : str, optional, defaults to "L1"
+        Loss function to be minimized. Should be one of "MSE", "huber", "L1", "soft_L1", or "cauchy".
+    huber_coef : float, optional, default to 20
+        Coefficient for Huber loss function.
+    min_refractors : int, optional, defaults to 1
+        Minimum number of refractors for the expected velocity model.
+    max_refractors : int, optional, defaults to 10
+        Maximum number of refractors for the expected velocity model.
+    find_weathering : bool, optional, defaults to False.
+        If True the minimum refractor size contraint for the expected weathering layer is removed.
+
+    Returns
+    -------
+    rv : RefractorVelocity or None
+        A near-surface velocity model with optimal number of refractors. Returns None if no velocity model is possible
+        for the given parameters.
+    """
+    #pylint: disable-next=import-outside-toplevel
+    from .refractor_velocity import RefractorVelocity  # avoid circulat import
+    rv = None
+    for refractor in range(min_refractors, max_refractors + 1):
+        min_refractor_size_vec = np.full(refractor, min_refractor_size)
+        if find_weathering:
+            min_refractor_size_vec[0] = 1
+        max_offset = max(offsets.max(), min_refractor_size * refractor)
+        rv_last = RefractorVelocity.from_first_breaks(offsets, times, init, bounds, refractor, max_offset,
+                                                      min_velocity_step, min_refractor_size_vec, loss, huber_coef)
+        n_points, _ = np.histogram(rv_last.offsets, bins=rv_last.piecewise_offsets)
+        # TODO: remove debug
+        if debug:
+            rv_last.plot(title=(f'loss: {rv_last.fit_result.fun:.6f}\nfind_weathering={find_weathering}'
+                                f'\nn_points: {n_points}'))
+        if not ((n_points > 1).all() and (rv is None or rv_last.fit_result.fun < rv.fit_result.fun)):
```

Comment:
If you don't want to add toleranceas @dpodvyaznikov suggested, at least simplify first part to 
```suggestion
        if (n_points < 1).any() and (rv is None or rv_last.fit_result.fun < rv.fit_result.fun)):
```